docs/developer/architecture.md

```diff
@@ -0,0 +1,101 @@
+# Architecture
+
+This page documents the technical architecture of `conda-pypi`, explaining
+how it integrates with conda and the internal organization of its components.
+
+## Plugin System Integration
+
+`conda-pypi` is implemented as a conda plugin using `conda`'s official plugin
+architecture. This design enables seamless integration with conda's existing
+workflows without requiring modifications to conda itself.
+
+The plugin registers several hooks with `conda`'s plugin system. The
+subcommand hook adds the `conda pypi` subcommand to conda through
+`conda_pypi.plugin.conda_subcommands()`, providing both `conda pypi install`
+for installing PyPI packages with conversion and `conda pypi convert` for
+converting PyPI packages without installing them.
+
+The plugin also registers three post-command hooks that extend conda's
+existing commands. The environment protection hook triggers after `install`,
+`create`, `update`, and `remove` commands to automatically deploy
+`EXTERNALLY-MANAGED` files that prevent accidental `pip` (or any other Python install tool) usage. This is
+implemented through `ensure_target_env_has_externally_managed()`.
+
+## Data Flow Architecture
+
+### Installation Flow
+
+```
+conda pypi install package
+         ↓
+CLI Argument Parsing
+         ↓
+Environment Validation
+         ↓
+Package Classification
+         ↓
+    VCS/Editable? ----Yes---→ Direct pip install
```

Comment:
```suggestion
    VCS/Editable? ----Yes---→ Direct pip install
```
We will end up reworking this part I guess, so just noting it here but not important now: I assume the `EXTERNALLY-MANAGED` file is also created afterwards, right?